### ISSUE TITLE
MathJax v3

### ISSUE THREAD

**Issue by benide** [kaushalmodi/ox-hugo#297]
This isn't a bug report, I just want to bring something to your attention in case it ends up mattering later.

MathJax v3 was pretty much a total rewrite. It speeds things up considerably, which is great. The way it is configured has changed, so it isn't as simple as changing 2.7.4 to 3.0.0 in your mathjax partial. I'll let you know if I get around to trying it out, and I'll let you know if any math that I use breaks from it. It'll be a little while because I'm trying to finish writing a thesis.

Here's a page with some of the details: https://docs.mathjax.org/en/latest/upgrading/v2.html

**Comment by kaushalmodi** [comment]
Thanks for this heads-up! I don't actively use mathjax. So I'll appreciate your input on this.

**Comment by kaushalmodi** [comment]
Please see https://github.com/kaushalmodi/hugo-bare-min-theme/issues/11. I have observed a regression in the parsing of (r), (c), (tm) by MathJax 3 vs MathJax 2.

**Comment by kaushalmodi** [comment]
Also the alignment of equations with equation numbers is not right. See https://ox-hugo.scripter.co/test/posts/equations-bf-escaping/

See more ox-hugo + MathJax tests at https://ox-hugo.scripter.co/test/search/?q=mathjax.

**Comment by benide** [comment]
The equation number thing is very strange. With the same config on my site, I copied and pasted that part of your `all-posts.org` over to my `.org` file and it does not happen. [See here, scroll down to "Tests"](https://www.bencide.com/blog/first-post). I'll poke around and see if I can make it happen somehow...

**Comment by kaushalmodi** [comment]
What is your Hugo version? And can you share the Markdown?

**Comment by benide** [comment]
v0.59.0-1DD0C69C/extended

Sorry for the delay, my remote repo wasn't github anymore (and I'm a newbie so this took a second) so I had to figure out how to add this back as a second remote. Here's the markdown link: https://github.com/benide/bencide.com/blob/master/content/blog/first-post.md

**Comment by kaushalmodi** [comment]
Some development..

I copy-pasted your first-post.md as temp.md in ox-hugo test site's content dir, and it rendered the equation numbers alright as you also see.

But when I removed the first equation block from that temp.md:

```
\begin{align}
  f(x) &= \begin{cases}
    1 & x \geq 0 \\\\\\
    0 & \text{otherwise}
  \end{cases}\label{eq1} \\\\\\
  \sum\_{k=0}^5g(x\_k) &= \\|h(y)\\|
\end{align}
```

, I see the below:

![image](https://user-images.githubusercontent.com/3578197/67705459-69aa8a80-f98d-11e9-9df1-e11dd0a3bdbc.png)

May be this helps you understand the root cause of the issue.

**Comment by kaushalmodi** [comment]
The first block "somehow helps fix the rendering of those equation numbers".

**Comment by benide** [comment]
Ah, I see. I just did some quick testing, looks like `\begin{cases}` is triggering it to load an extra module. Let me check which one that loads and I'll update the config to manually load it. I guess I should also put in an issue in MathJax.

**Comment by benide** [comment]
Not exactly what I thought was going on, but it has been fixed: mathjax/MathJax-src#343. I assume this will be in 3.0.1. Doesn't look like there is an easy way to make this work with chtml output in the meantime. Two options:

1. Revert to MathJax v2 (kaushalmodi/hugo-bare-min-theme#12) until 3.0.1 comes out
2. Switch to the svg renderer (load `tex-svg.js` instead of `tex-chtml.js`)

I think 1 is better personally (though, for my self, I'll just use 3.0.0 with chtml and deal with it haha)

**Comment by kaushalmodi** [comment]
OK, thank you for your time with this debug; I'll revert to MathJax v2 until 3.0.1 comes out. I'll leave this issue open for that future upgrade to 3.0.1.
